packages/debug/src/browser/debug-session-manager.ts

```diff
@@ -424,11 +425,22 @@ export class DebugSessionManager {
             return this.doPostTaskAction(`Could not run the task '${taskName}'.`);
         }
 
-        const code = await this.taskService.getExitCode(taskInfo.taskId);
-        if (code === 0) {
+        const getExitCodePromise: Promise<TaskEndedInfo> = this.taskService.getExitCode(taskInfo.taskId).then(result =>
+            ({ taskEndedtype: TasksEndedTypes.TaskExited, value: result }));
+        const isBackgroundTaskEndedPromise: Promise<TaskEndedInfo> = this.taskService.isBackgroundTaskEnded(taskInfo.taskId).then(result =>
+            ({ taskEndedtype: TasksEndedTypes.BackgroundTaskEnded, value: result }));
+
+        // After start running the task, we wait for the task proccess to exit and if it is a background task, we also wait for a feedback
+        // that a background task is active, as soon as one of the promises fulfills, we can continue and analyze the results.
+        const taskEndedInfo: TaskEndedInfo = await Promise.race([getExitCodePromise, isBackgroundTaskEndedPromise]);
+
+        if (taskEndedInfo.taskEndedtype === TasksEndedTypes.BackgroundTaskEnded && taskEndedInfo.value) {
```

Comment:
[minor] please rename to `taskEndedType` following the [guidelines](https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#camelcase-var).

Reply:
fixed, done